lib/nsq/connection.rb

```diff
@@ -32,6 +32,7 @@ def initialize(opts = {})
       @msg_timeout = opts[:msg_timeout] || 60_000 # 60s
       @max_in_flight = opts[:max_in_flight] || 1
       @ssl_context = opts[:ssl_context]
+      @tls_v1 = !!opts[:tls_v1]
       validate_ssl_context! if @ssl_context
```

Comment:
@alieander Good point. The weird `{tls_v1: false, ssl_context: {...}}` case strikes again! I'm thinking maybe we skip validation there and throw some warning to the console that they supplied an ssl context but disabled tls. Thoughts on that?